metadata/src/test/java/edu/unc/lib/dl/util/DepositStatusFactoryIT.java

```diff
@@ -0,0 +1,139 @@
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.util;
+
+import static org.junit.Assert.*;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositAction;
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositField;
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositState;
+import redis.clients.jedis.JedisPool;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration({"/spring-test/cdr-client-container.xml"})
+public class DepositStatusFactoryIT {
+	
+	private DepositStatusFactory factory;
+	@Autowired
+	private JedisPool jedisPool;
+	
+	@Before
+	public void init() {
+		factory = new DepositStatusFactory();
+		factory.setJedisPool(jedisPool);
+	}
+
+	@Test
+	public void testAddThenGetManifest() {
+		final String uuid = UUID.randomUUID().toString();
+		final String filename1 = "bagit.txt";
+		final String filename2 = "manifest-md5.txt";
+		
+		factory.addManifest(uuid, filename1);
+		factory.addManifest(uuid,  filename2);
+		List<String >filenames = factory.getManifestURIs(uuid);
+		
+		assertTrue(filenames.size() == 2);
+		assertEquals(filename1, filenames.get(0));
+		assertEquals(filename2, filenames.get(1));
+		
+		factory.delete(uuid);
+	}
+	
+	@Test
+	public void testAddRemoveSupervisorLock() {
+		final String uuid = UUID.randomUUID().toString();
+		String owner1 = "owner1";
+		String owner2 = "owner2";
+		assertTrue(factory.addSupervisorLock(uuid, owner1));
+		assertFalse(factory.addSupervisorLock(uuid, owner2));
+		factory.removeSupervisorLock(uuid);
+		assertTrue(factory.addSupervisorLock(uuid, owner2));
+		
+		factory.delete(uuid);
+	}
+	
+	@Test
+	public void testSetStateGetState() {
+		final String uuid = UUID.randomUUID().toString();
+		factory.setState(uuid, DepositState.queued);
+		assertEquals(DepositState.queued, factory.getState(uuid));
+		
+		factory.delete(uuid);
+	}
+	
+	@Test
+	public void testSetGetDeleteExpireFail() throws InterruptedException {
+		final String uuid = UUID.randomUUID().toString();
+		factory.set(uuid, DepositField.contactName, "Boxy");
+		factory.set(uuid, DepositField.fileName, "boxys_file.txt");
+		
+		Map<String,String> status = factory.get(uuid);
+		assertTrue(status.size() == 2);
+		assertEquals("Boxy", status.get(DepositField.contactName.toString()));
+		assertEquals("boxys_file.txt", status.get(DepositField.fileName.toString()));
+		
+		factory.deleteField(uuid, DepositField.fileName);
+		status = factory.get(uuid);
+		assertNull(status.get(DepositField.fileName.toString()));
+		assertTrue(status.size() == 1);
+		
+		final String uuid2 = UUID.randomUUID().toString();
```

Comment:
Might be clearer if this where split into a separate test?  One tests delete, the other tests expire.